sale_layout_hidden/models/sale_layout.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+
+
+class SaleLayoutCategory(models.Model):
+    _inherit = 'sale.layout_category'
+
+    hidden = fields.Boolean(default=False)
+
+    @api.model
+    def create(self, values):
+        if values.get('hidden'):
+            values = values.copy()
```

Comment:
@angelmoya we are changing the values (see line 17 below) => this avoids changing the dictionarly in the caller's context as dictionaries are mutable.